actions/bump-and-release-cloud-plugin/action.yaml

```diff
@@ -0,0 +1,308 @@
+  Bumps the version of a cloud plugin and tags the release. This action only builds Go binaries for the linux platform
+  as that is the only platform that Grafana Cloud uses.
+
+inputs:
+  default-branch:
+    description: |
+      The default branch of the repository. This is used to checkout the repository.
+    default: main
+    required: true
+  github-token:
+    description: |
+      A GitHub token used to commit the version bump and create a tag. This will need to have write access to the
+      default branch of the repository.
+    required: true
+  gcs-bucket:
+    description: GCS bucket to upload too
+    required: true
+  gcs-service-account-creds:
+    description: Credentials JSON to log into GCP and upload plugin to bucket
+    required: true
+  package-manager:
+    description: |
+      The package manager to use for building the plugin. This can be either `yarn` or `npm`.
+    default: yarn
+    required: true
+  release-level:
+    description: |
+      The level of the release. This can be one of `major`, `minor`, `patch`, or `prerelease`.
+      See https://semver.org/ for more information.
+    required: true
+  signing-token:
+    description: |
+      A Grafana access policy token to sign the plugin with.
+      See https://grafana.com/developers/plugin-tools/publish-a-plugin/sign-a-plugin#generate-an-access-policy-token
+    required: true
+  update-changelog:
+    description: |
+      Whether to update the changelog with the new version. When updating will replace the `## Unreleased` section
+      with `## Unreleased\n\n## v*.*.*`. Giving the top of the changelog a new unreleased section.
+    default: 'false'
+    required: true
+outputs:
+  version:
+    description: The semantic version that the package.json was bumped too
+    value: ${{ steps.output-version.outputs.version }}
+  version-tag:
+    description: The tag that was created for the version
+    value: ${{ steps.output-version.outputs.version-tag }}
+
+runs:
+  using: composite
+  steps:
+    - name: Specify GOPRIVATE environment variable
+      shell: bash
+      run: |
+        echo "GOPRIVATE=github.com/${{ github.repository_owner }}/*" >> $GITHUB_ENV
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        ref: ${{ inputs.default-branch }}
+        token: ${{ inputs.github-token }}
+        fetch-depth: 0
+
+    - name: Setup credentials to access private repositories
+      shell: bash
+      run: git config --global url.https://${{ inputs.github-token }}@github.com/.insteadOf https://github.com/
+
+    - name: Setup Node.js environment
+      uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4.0.2
+      with:
+        node-version: lts/iron
+        check-latest: true
+        cache: ${{ inputs.package-manager }}
+
+    - name: Setup Go environment
+      uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+      with:
+        go-version: '1.21'
```

Comment:
Is this version something that could vary from plugin to plugin or that could be extracted from the context where the Action is used?

Reply:
Wow https://github.com/grafana/plugin-validator has come a long way since I looked at the repo. This should totally be customizable!